apps/astarte_housekeeping/lib/astarte_housekeeping/engine.ex

```diff
@@ -176,4 +219,20 @@ defmodule Astarte.Housekeeping.Engine do
 
     update_valid?
   end
+
+  defp merge_realm_status(old_realm, %{device_registration_limit: nil} = update_attrs) do
+    attrs = Map.drop(update_attrs, [:device_registration_limit])
+    merge_realm_status(old_realm, attrs)
+  end
+
+  defp merge_realm_status(old_realm, %{device_registration_limit: :remove_limit} = update_attrs) do
+    attrs = Map.drop(update_attrs, [:device_registration_limit])
+    merge_realm_status(%{old_realm | device_registration_limit: nil}, attrs)
+  end
+
+  defp merge_realm_status(old_realm, update_attrs) do
+    Map.merge(old_realm, update_attrs, fn _k, old_value, new_value ->
+      if new_value == nil, do: old_value, else: new_value
```

Comment:
More compact:
`new_value || old_value`

Reply:
More compact, but at what cost? We are sacrificing clarity (the new_value *might* be the atom nil) for 2 lines of code that feel like `new_value` and `old_value` are booleans. Even if `||` is idiomatic Elixir, I would prefer the explicit comparison, in order to better convey the type information